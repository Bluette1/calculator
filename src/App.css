#root {
  margin: 5% 5%;
}

@media (min-width: 992px) {
  #root {
    width: 700px;
    height: 800px;
    text-align: center;
  }
}
