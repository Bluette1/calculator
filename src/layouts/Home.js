import React from 'react';

const Home = () => (
  <div className="home">
    <h3>Welcome to our page!</h3>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Quas,
      veniam unde doloremque eius aperiam aliquid, repellat consectetur
      alias voluptatem necessitatibus architecto error possimus et,
      ducimus autem corporis repudiandae voluptatibus! Quam?
    </p>
    <p>
      Lorem ipsum, dolor sit amet consectetur adipisicing elit.
      Ipsum fugit dolores ex dicta dolore mollitia vitae sed et,
      nam praesentium sapiente autem vel labore dignissimos, qui harum,
      repellendus sit perspiciatis.
    </p>
  </div>
);

export default Home;
