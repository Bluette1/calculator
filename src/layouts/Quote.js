import React from 'react';

const Quote = () => (
  <div className="quote">
    <h3>Quote of the day</h3>
    <p>
      Mathematics is not about numbers, equations, computations, or algorithms:
      it is about understanding. â€“William Paul Thurston
    </p>
  </div>
);

export default Quote;
