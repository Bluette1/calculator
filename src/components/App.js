import React from 'react';
import '../App.css';
import ButtonGroups from './ButtonGroups';
import buttonGroups from '../helpers/buttonGroups';
import Display from './Display';
import calculate from '../logic/calculate'

const App = () => (
  <>
    <Display />
    <ButtonGroups groups={buttonGroups()} />
  </>
);

export default App;
