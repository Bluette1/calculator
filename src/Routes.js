import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import App from './components/App';
import Quote from './layouts/Quote';
import Home from './layouts/Home';

const Routes = () => (
  <BrowserRouter>
    <Switch>
      <Route exact path="/calculator" component={App} />
      <Route exact path="/quote" component={Quote} />
      <Route exact path="/" component={Home} />
    </Switch>
  </BrowserRouter>
);

export default Routes;
