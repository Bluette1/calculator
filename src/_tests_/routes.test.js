import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Route } from 'react-router-dom';
import Routes from '../Routes';
import App from '../components/App';
import Quote from '../layouts/Quote';
import Home from '../layouts/Home';
import NotFound from '../layouts/NotFound';

configure({ adapter: new Adapter() });

let pathMap = {};
describe('routes using array of routers', () => {
  beforeAll(() => {
    const component = shallow(<Routes />);
    pathMap = component.find(Route).reduce((map, route) => {
      const routeMap = map;
      const routeProps = route.props();
      routeMap[routeProps.path] = routeProps.component;
      return routeMap;
    }, {});
  });
  it('should show Home component for / route', () => {
    expect(pathMap['/']).toBe(Home);
  });
  it('should show App component for /calculator route', () => {
    expect(pathMap['/calculator']).toBe(App);
  });
  it('should show Quote component  for /quote route', () => {
    expect(pathMap['/quote']).toBe(Quote);
  });
  it('should show NotFound o component for any route that is not defined', () => {
    expect(pathMap.undefined).toBe(NotFound);
  });
});
