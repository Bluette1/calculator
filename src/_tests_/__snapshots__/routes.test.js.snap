// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes using array of routers should show App component for /calculator route 1`] = `[Function]`;

exports[`routes using array of routers should show NotFound component for any route that is not defined 1`] = `[Function]`;

exports[`routes using array of routers should show Quote component  for /quote route 1`] = `[Function]`;
